### PR TITLE
misc typing mainly for ninjabackend

### PR DESCRIPTION
**Pull request by eli-schwartz**
Minus a couple hundred mypy errors in ninjabackend.py

### CODE REVIEW

**Review by tristan957 — APPROVED** [review]
Nice work!

File: mesonbuild/backend/ninjabackend.py [hunk 1]
```diff
@@ -485,13 +485,14 @@ def __init__(self, build: T.Optional[build.Build], interpreter: T.Optional[Inter
         self.name = 'ninja'
         self.ninja_filename = 'build.ninja'
         self.fortran_deps = {}
-        self.all_outputs = {}
+        self.all_outputs: T.Dict[str, Literal[True]] = {}
```
Comment (xclaesse) [inline mesonbuild/backend/ninjabackend.py:488]: Shouldn't that be changed to `T.Set[str]` while at it?

File: mesonbuild/msetup.py [hunk 2]
```diff
@@ -174,15 +174,15 @@ def validate_dirs(self, dir1: str, dir2: str, reconfigure: bool, wipe: bool) ->
         return src_dir, build_dir
 
     # See class Backend's 'generate' for comments on capture args and returned dictionary.
-    def generate(self, capture: bool = False, vslite_ctx: dict = None) -> T.Optional[dict]:
+    def generate(self, capture: bool = False, vslite_ctx: T.Optional[dict] = None) -> T.Optional[dict]:
```
Comment (xclaesse) [inline mesonbuild/msetup.py:177]: Can't we type what that dict contains as well?

File: mesonbuild/backend/ninjabackend.py (hunk 1 above)
Comment (eli-schwartz) [reply mesonbuild/backend/ninjabackend.py:488]: You mean change the code to use a set instead of a True-keyed dict?

Comment (xclaesse) [reply mesonbuild/backend/ninjabackend.py:488]: Yes, unrelated to this typing PR, but if we don't cleanup that now we'll never do it :P

File: mesonbuild/msetup.py (hunk 2 above)
Comment (eli-schwartz) [reply mesonbuild/msetup.py:177]: Someone could, certainly, but it is always None except inside the VS backend -- this is specific to the new genvslite backend. And quite frankly I do not want to know how that works or investigate how to type annotate it... I've gone a bit off the whole idea of that backend lol.

And yes, I know that before the genvslite code was merged I went through and added quite a bit of typing to vs2010backend.py

File: mesonbuild/backend/ninjabackend.py (hunk 1 above)
Comment (eli-schwartz) [reply mesonbuild/backend/ninjabackend.py]: Should be fixed.

**Comment by tristan957** [comment]
the second commit has a space in between `ninja` and `backend` if you care

**Pull request merged**